rs/replicated_state/src/canister_state/queues/message_pool/tests.rs

```diff
@@ -380,14 +406,20 @@ fn test_equality() {
     assert_eq!(pool, other_pool);
 
     // Pop the same message from either pool.
-    assert!(pool.take(id1).is_some());
-    assert!(other_pool.take(id1).is_some());
+    assert!(pool.take(ref1).is_some());
+    assert!(other_pool.take(ref1).is_some());
     // The two pools should still be equal.
     assert_eq!(pool, other_pool);
 
     // Shed a message from either pool.
-    assert_eq!(id2, pool.shed_largest_message().unwrap().0);
-    assert_eq!(id2, other_pool.shed_largest_message().unwrap().0);
+    assert_eq!(
+        SomeReference::Inbound(ref2),
+        pool.shed_largest_message().unwrap().0
+    );
```

Comment:
Personal preference: you write this using `assert_matches` without `unwrap().0`. And other places too.

Reply:
It works, but it's more verbose:
```rust
    assert_matches!(
        pool.shed_largest_message(),
        Some((SomeReference::Inbound(r), _)) if r == ref2
    );
```

Reply:
I was hoping it would work without an `if` clause. I tend to prefer a triggering `assert` rather than an `unwrap()` inside of it but feel free to leave it as it is.